src/rust/engine/fs/src/lib.rs

```diff
@@ -222,6 +216,31 @@ impl PathStat {
       &PathStat::File { ref path, .. } => path.as_path(),
     }
   }
+
+  ///
+  /// Sort and ensure that there were no duplicate entries.
+  ///
+  /// TODO: audit if this is even necessary? expand_globs() already sorts and dedupes, but, are
+  /// there other callers of this where we can't be confident the sorting happened? Maybe, we
+  /// should add a light wrapper around `Vec<PathStat>` that ensures it's sorted and deduped, as
+  /// we're now using the type a lot.
+  pub fn normalize_path_stats(mut path_stats: Vec<PathStat>) -> Result<Vec<PathStat>, String> {
```

Comment:
This was extracted out of `snapshot.rs`. I thought I was going to need this for `FileListing`, but turns out we don't because `expand_globs()` already sorts and dedupes.

I kept this extracted, though, because it allows deduplicating two other call sites.